src/layout.rs

```diff
@@ -570,8 +570,8 @@ fn try_split(area: Rect, layout: &Layout) -> Result<Rc<[Rect]>, AddConstraintErr
     let mut results = elements
         .iter()
         .map(|element| {
-            let start = *changes.get(&element.start).unwrap_or(&0.0);
-            let end = *changes.get(&element.end).unwrap_or(&0.0);
+            let start = changes.get(&element.start).unwrap_or(&0.0).round();
+            let end = changes.get(&element.end).unwrap_or(&0.0).round();
```

Comment:
I'm not sure this works in the general case. Can we add a comment about this? Something like `// TODO: explore corner cases for when .round() results in gaps`

Reply:
Round can't leave gaps as end[i-1] always = start[i] (required constraint). If that constraint can not be met, then the split method will panic before this point.

This suggests that perhaps we could make this code explicitly gap free by only using the end values as variables and dropping the start variables.